src/hooks/BookAppointment/useAppointmentCalendar.hook.tsx

```diff
@@ -1,54 +1,211 @@
-import { useState } from 'react';
+import { useState, useEffect, useMemo } from 'react';
 import { DayClickEventHandler } from 'react-day-picker';
 import { addMonths } from 'date-fns';
-
-type Props = {
+import { appointmentApi } from 'services/BookAppointmetService';
+import { BLUE, HINT } from '@constants/colors';
+import { TIME_OPTIONS } from '@constants/other';
+interface Props {
   onDayClick: (day: Date) => void;
-};
+  specialization: number;
+  setAvalibleTimeRange: React.Dispatch<React.SetStateAction<any>>;
+  selectedDay: Date | null;
+  setSelectedDay: React.Dispatch<React.SetStateAction<Date | null>>;
+  setSelectedDate: React.Dispatch<React.SetStateAction<Date | null>>;
+  setFormattedDate: React.Dispatch<React.SetStateAction<string>>;
+}
+
+interface FreeSlotProps {
+  doctor: {
+    id: number;
+    role: string;
+    specialization: number;
+  };
+  start: string;
+  end: string;
+  id: number;
+  title: string;
+  uuid: string;
+}
 
-const useAppointmentCalendarHook = ({ onDayClick }: Props) => {
+const useAppointmentCalendarHook = ({
+  onDayClick,
+  specialization,
+  setAvalibleTimeRange,
+  selectedDay,
+  setSelectedDay,
+  setSelectedDate,
+  setFormattedDate,
+}: Props) => {
   const today = new Date();
-  const [selectedDay, setSelectedDay] = useState<Date>(today);
   const nextMonth = addMonths(new Date(), 0);
-  const [month, setMonth] = useState(nextMonth);
-  const currentStyle = { backgroundColor: '#4579EE' };
+  const [month, setMonth] = useState<Date>(nextMonth);
+  const [showCalendar, setShowCalendar] = useState<boolean>(false);
+  const [freeSlots, setFreeSlots] = useState<FreeSlotProps[]>([]);
+  const currentStyle = { backgroundColor: `${BLUE}` };
+  const bookedStyle = { color: `${HINT}` };
+  const { data } = appointmentApi.useGetSpecializationByIdQuery(specialization);
+
+  useEffect(() => {
+    if (specialization !== undefined && data) {
+      setFreeSlots(data);
+    }
+  }, [specialization, data]);
+
+  useEffect(() => {
+    if (freeSlots.length === 0) {
+      setSelectedDay(null);
+    }
+  }, [freeSlots]);
+
+  const ThreeMonthPeriod = useMemo(() => {
```

Comment:
why constant from upper case? all constants from lower case and this from upper case